src/modules/uORB/uORBDeviceMaster.cpp

```diff
@@ -141,12 +141,6 @@ int uORB::DeviceMaster::advertise(const struct orb_metadata *meta, bool is_adver
 
 				if (existing_node != nullptr &&
 				    (!existing_node->is_advertised() || is_single_instance_advertiser || !is_advertiser)) {
-					if (is_advertiser) {
-						/* Set as advertised to avoid race conditions (otherwise 2 multi-instance advertisers
-						 * could get the same instance).
-						 */
-						existing_node->mark_as_advertised();
```

Comment:
`px4_open() -> DeviceNode::open()` This has already been done and there is no need to deal with it again.

Reply:
Check the comment, there's a race condition when removing this.